### PR TITLE
Fix for Issue #78

### PR DESCRIPTION
**Pull request by truemedian**
Quick fix that should handle Issue #78 in most cases.

### CODE REVIEW

File: libs/client/EventHandler.lua [hunk 1]
```diff
@@ -284,11 +284,16 @@ function EventHandler.GUILD_MEMBER_UPDATE(d, client)
...
 
+local presence_fix = {}
 function EventHandler.GUILD_MEMBER_REMOVE(d, client)
 	local guild = client._guilds:get(d.guild_id)
 	if not guild then return warning(client, 'Guild', d.guild_id, 'GUILD_MEMBER_REMOVE') end
 	local member = guild._members:_remove(d)
 	guild._member_count = guild._member_count - 1
+	if client._options.cacheAllMembers then
+		presence_fix[guild.id] = presence_fix[guild.id] or {}
+		presence_fix[guild.id][d.user.id] = true
```
Comment (SinisterRectus) [inline libs/client/EventHandler.lua]: For consistency, this should be nil'd on GUILD_MEMBER_ADD.

File: libs/client/EventHandler.lua [hunk 2]
```diff
@@ -426,6 +431,14 @@ function EventHandler.PRESENCE_UPDATE(d, client) -- may have incomplete data
 		else
 			member = guild._members:get(d.user.id)
 			if not member then
+				if client._options.cacheAllMembers then
+					presence_fix[guild.id] = presence_fix[guild.id] or {}
+					if presence_fix[guild.id][d.user.id] then
+						presence_fix[guild.id][d.user.id] = nil
+						client:debug('Ignoring PRESENCE_UPDATE for user \'%s\' in guild \'%s\'', d.user.id, guild.id)
+						return
```
Comment (SinisterRectus) [inline libs/client/EventHandler.lua]: Move return up for a proper tail-call.

File: libs/client/EventHandler.lua [hunk 3]
```diff
@@ -284,11 +284,16 @@ function EventHandler.GUILD_MEMBER_UPDATE(d, client)
...
 end
 
+local presence_fix = {}
 function EventHandler.GUILD_MEMBER_REMOVE(d, client)
 	local guild = client._guilds:get(d.guild_id)
 	if not guild then return warning(client, 'Guild', d.guild_id, 'GUILD_MEMBER_REMOVE') end
 	local member = guild._members:_remove(d)
 	guild._member_count = guild._member_count - 1
+	if client._options.cacheAllMembers then
+		presence_fix[guild.id] = presence_fix[guild.id] or {}
```
Comment (SinisterRectus) [inline libs/client/EventHandler.lua]: Might want to use an auto-expanding table. See listenersMeta in Emitter.lua.

File: libs/client/EventHandler.lua [hunk 4]
```diff
@@ -426,6 +431,14 @@ function EventHandler.PRESENCE_UPDATE(d, client) -- may have incomplete data
 		else
 			member = guild._members:get(d.user.id)
 			if not member then
+				if client._options.cacheAllMembers then
+					presence_fix[guild.id] = presence_fix[guild.id] or {}
```
Comment (SinisterRectus) [inline libs/client/EventHandler.lua]: Might want to use an auto-expanding table. See listenersMeta in Emitter.lua.

**Comment by truemedian** [comment]
was a quick fix, will move to integrate it more to the lib

**Comment by SinisterRectus** [comment]
Were you able to test this to see if it fixes the issue?

**Comment by truemedian** [comment]
Yes, after receiving PRESENCE_UPDATE it does ignore the event for that member

**Comment by SinisterRectus** [comment]
Does the member count stay accurate?

**Comment by truemedian** [comment]
Hmm, good question, will test

**Comment by truemedian** [comment]
After some testing on multiple guilds, I haven't had it fail; its giving accurate member counts
should work in all cases

**Pull request merged**